.github/workflows/ci.yml

```diff
@@ -38,27 +39,23 @@ jobs:
       - uses: actions/checkout@v4
       - name: Set up Python
         uses: actions/setup-python@v5
-        with:
-          python-version: ${{ matrix.python-version }}
-          cache: "pip"
-      - name: Install dependencies
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
+      - name: Install dependencies after creating a virtual environment
         run: |
-          python -m pip install --upgrade pip
-          python -m pip install -r requirements.txt -r requirements-dev.txt
-          python -m pip install pytest mypy
-      - name: Build
-        run: python -m pip install -e .
+          uv venv
+          uv sync --group all
```

Comment:
I have removed the 'Build' step because `uv sync` installs the editable version of tqec by default.